File: Strings/StringCompression.cpp
#include<bits/stdc++.h>
using namespace std;

int compress(vector<char> & s) {
    int n = s.size();
    int count = 1;
    int i = 0;
    int k = 0;
    for(i=0; i<n-1; i++) {
        if(s[i] != s[i+1]) {
            if(count > 1) {
                s[k] = s[i];
                k++;
                string count_str = to_string(count);
                for(int j=0; j<count_str.size(); j++) {
                    s[k] = count_str[j];
                    k++;
                }
                count = 1;
            } else {
                s[k] = s[i];
            }
        }
        else {
            count++;
        }
    }
    
    if(count > 1) {
        s[k] = s[i];
        k++;
        string count_str = to_string(count);
        for(int j=0; j<count_str.size(); j++) {
            s[k] = count_str[j];
            k++;
        }
    }
    else {
        s[k] = s[i];
        k++;
    }
    
    return k;
}

int main() {
    vector<char> v = {'a','a','b','b','c','c','c','c','c','c','c','c','c','c','c','c','d','d'};
    int len = compress(v);
    for(int i=0; i<len; i++) {
        cout<<v[i];
    }
}

File: Strings/StringCompression.js
function StringCompression(chars) {
  let n = chars.length;
  let count = 1;
  let k = 0;
  let i = 0;
  for (i = 0; i < n - 1; i++) {
    if (chars[i] !== chars[i + 1]) {
      let ch = chars[i];
      if (count === 1) {
        chars[k] = ch;
        k++;
      } else {
        chars[k] = ch;
        k++;
        let count_str = String(count);
        for (let j = 0; j < count_str.length; j++) {
          chars[k] = count_str[j];
          k++;
        }
      }
      count = 1;
    } else {
      count++;
    }
  }

  let ch = chars[i];
  if (count === 1) {
    chars[k] = ch;
    k++;
  } else {
    chars[k] = ch;
    k++;
    let count_str = String(count);
    for (let j = 0; j < count_str.length; j++) {
      chars[k] = count_str[j];
      k++;
    }
  }
  return k;
}

const length = StringCompression(["a", "a", "b", "b", "c", "c", "c"]);
console.log(length);
